crates/cheatcodes/spec/src/vm.rs

```diff
@@ -166,6 +244,15 @@ interface Vm {
     #[cheatcode(group = Evm, safety = Safe)]
     function accesses(address target) external returns (bytes32[] memory readSlots, bytes32[] memory writeSlots);
 
+    /// Record all account accesses as part of CREATE, CALL or SELFDESTRUCT opcodes in order,
+    /// along with the context of the calls
+    #[cheatcode(group = Evm, safety = Safe)]
+    function startStateDiffRecording() external;
+
+    /// Returns an ordered array of all account accesses from a `vm.startStateDiffRecording` session.
+    #[cheatcode(group = Evm, safety = Safe)]
+    function stopAndReturnStateDiff() external returns (AccountAccess[] memory accesses);
```

Comment:
nit: from UX perspective, would it be useful to have a "return accesses but don't stop", or alternatively "stop but don't clear existing accesses"? Does the current behaviour mimic the other record cheats?

Reply:
This behavior does mimic other cheats like `getRecordedLogs`, which also stops log recording. I decided to rename it a bit differently so that it's clear that it also stops recording.
Other record cheats such as `record`/ `accesses` and `recordLogs` / `getRecordedLogs` don't have functions that  only stop without returning anything either.
I suppose users can always call `stopAndReturnStateDIff` without using the returned state diffs if they just want to stop recording.